event_session/models/event.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 David Vidal<david.vidal@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class EventEvent(models.Model):
+    _inherit = 'event.event'
+
+    session_ids = fields.One2many(
+        comodel_name='event.session',
+        inverse_name='event_id',
+        string='Sessions',
+    )
+    sessions_count = fields.Integer(
+        compute='_compute_sessions_count'
+    )
+
+    @api.multi
+    def _compute_sessions_count(self):
+        for event in self:
+            event.sessions_count = len(event.session_ids)
+
+
+class EventRegistration(models.Model):
+    _inherit = 'event.registration'
+
+    event_sessions_count = fields.Integer(
+        related='event_id.sessions_count',
+        readonly=True,
+    )
+    session_id = fields.Many2one(
+        comodel_name='event.session',
+        string='Session',
+        ondelete='set null',
```

Comment:
This should be `ondelete='restrict'`. You cannot remove a session if you have registration on it.